lib/tools/profiler.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const v8Profiler = require('v8-profiler');
+const EventEmitter = require('events');
+
+/**
+ * The Profiler
+ */
+class Profiler extends EventEmitter {
+
+  /**
+   * Create a new Profiler.
+   *
+   * @params {Object} options - Object with the profiler configuration options
+   *                  interval: The amount in milliseconds to wait before taking snapshots
+   *                  tag: custom identifier
+   *                  extension: The extension of the file. It must be of the type:
+   *                             .cpuprofile, .heapsnapshot, .heapprofile, or .heaptimeline
+   */
+  constructor(options) {
+    super();
+    this.extension = this.sanitize(options.extension || '');
```

Comment:
I think it makes sense to sanitize the extension option, the empty string is just a last resort fallback to not trigger an exception as undefined would not have a function named `replace`:
```
  sanitize(string) {
    return string.replace(/[&@:,$=+?;\s^`><{}[\]#%~|\\"]/ig, ''); // here
  }
```